### ISSUE TITLE
Raspberry Pi 4 not forcing OpenGL ES (Raspbian/Gentoo)

### ISSUE THREAD

**Issue by zRevengee** [Tinob/Ishiiruka#139]
in normal dolphin i can play some games by forcing GLES using
PreferGLES=True (Raspberry Pi doesn't have OpenGL support atm)

if i type this in the GFX.ini file it does nothing. (running on debian raspbian)
if ishiiruka will run on raspberry for sure we can get some games to full speed

**Comment by Tinob** [comment]
please test the latest version just uploaded some fixes to allow gles support.

**Comment by zRevengee** [comment]
i'm building it right now, thanks for the fast answer, i'll keep you updated

**Comment by zRevengee** [comment]
same issue, PreferGLES = True doesn't work
i have to set this to on? option(USE_EGL "Enables EGL OpenGL Interface" OFF)

**Comment by zRevengee** [comment]
with EGL enabled Ishiruuka crash when starting a game
realloc(): invalid old size
Aborted

**Comment by zRevengee** [comment]
it works, thanks man

**Comment by Tinob** [comment]
how is the performance?

**Comment by zRevengee** [comment]
Mario kart games have weird vertex explosions, luigi's mansion is full speed but with graphical glitches for the vacuum, other games are fine, melee full speed, dragon bal z budkai tenkaichi 3 fullspeed, brawl is full speed, super monkey ball is fullspeed, in nfs games there are serious fps issues, on dolphin i can play these games but on ishiiruka idk why these games runs like at 5 fps

**Comment by Tinob** [comment]
nice info will try to test them on my pi to see if i can debug the performance loss

**Comment by zRevengee** [comment]
you have to set screen res on 640x480 on raspbian with chroot
i'm posting a video on youtube, i will link it when it finish

**Comment by zRevengee** [comment]
Pi is OC 2147 Mhz CPU + 750Mhz on GPU

**Comment by zRevengee** [comment]
[https://youtu.be/p0GZiriEzbk](url)

**Comment by rhunjadi** [comment]
My Pi is OC 2100 Mhz CPU + 600Mhz on GPU... with the latest Mesa drivers (19.3.2)....I'm running the games over gentoo64...the games run slower than the ones on your video...So it looks like MESA drivers didn't give any accleration in games...

Could there be any optimizations in emulator? or is the RP4 just to weak to run on a deccent level?

**Comment by zRevengee** [comment]
You are wrong, mesa driver 19.3 are GLES 2.0 without ext frame buffer support, you need a raspbian build or the latest supreme retropie, with mesa 19.3 with ext frame buffer support, just 10 days ago the devs of mesa announced mesa 20 with GLES 3.1 , we’re currently testing a custom build with ext frame buffer support and it’s fairly verter, if you check my latest video you will see improvements in mesa 19(custom)

**Comment by zRevengee** [comment]
Don’t use gentoo, chroot raspbian or supreme is better optimized, i said before that mesa 19.3 on gentoo idk how it show 3.0 but it uses 2.0 instead, with the other os you should not have any issues

**Comment by rhunjadi** [comment]
Thank you for your anwser!

Hmmm I'm confused....because on the last update of Gentoo64 the command glxinfo shows 
OpenGL version 19.3.2

**Comment by rhunjadi** [comment]
Just out of curiosity I tried it on the latest gentoo64 (feb 2020)...and latest Mesa 20.0.0-rc2 drivers.
Everything compiles fine....but when I try to run a game I get this WARNING window:

Failed to compile vs shader: /home/demouser/.local/share/ishiiruka/Dump/bad_vs_0000.txt
Debug info (VMware, Inc., llvmpipe (LLVM 9.0.1, 128 bits), 3.3 (Core Profile) Mesa 20.0.0-rc2):
0:1(10): error: GLSL ES 3.20 is not supported. Supported versions are: 1.10, 1.20, 1.30, 1.40, 1.50, 3.30, 1.00 ES, and 3.00 ES

any help?
